packages/ui/src/composables/useTextColor.ts

```diff
@@ -0,0 +1,20 @@
+import { computed, getCurrentInstance, Ref, unref } from 'vue'
+import { useColors } from '../services/color-config/color-config'
+
+export const useTextColor = (componentColor: string | Ref<string>, isTransparent: boolean | Ref<boolean> = false) => {
+  const { props } = getCurrentInstance()!
+  const { getColor, getTextColor } = useColors()
+
+  const textColorComputed = computed(() => {
+    const hexColor = getColor(unref(componentColor))
+    const textColorProp = props.textColor as string
+
+    if (unref(isTransparent)) {
+      return hexColor
+    }
+
+    return props.textColor ? getColor(textColorProp) : getColor(getTextColor(hexColor))
```

Comment:
```suggestion
    return props.textColor ? getColor(props.textColor) : getColor(getTextColor(hexColor))
```

Reply:
We're losing TS support this way. I think we can make this type of refactoring after we discuss this [PR](https://github.com/epicmaxco/vuestic-ui/pull/1775).